### PR TITLE
Upgrade Jest to v28

### PR DESCRIPTION
**Pull request by jeremywiebe**
## Summary:

I was investigating the radio widget test failures. They were super slow (4s for some of the failing tests) and I went down a rabbit hole profiling it. Some of the forum posts I found referenced JSDOM. I decided to see how things would fare on the new Jest 28. And low and behold, all radio tests pass now. Plus they're much faster. 

Issue: FEI-4497

## Test plan:

```
$ yarn jest radio_test.jsx
 PASS  packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx
 PASS  packages/perseus/src/widgets/__tests__/radio_test.jsx (16.057 s)

Test Suites: 2 passed, 2 total
Tests:       91 passed, 91 total
Snapshots:   13 passed, 13 total
Time:        16.869 s, estimated 21 s
Ran all test suites matching /radio_test.jsx/i.
✨  Done in 18.04s.
```

### CODE REVIEW

File: packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap [hunk 1]
```diff
@@ -386,7 +386,7 @@ exports[`renderer snapshots correct answer: correct answer 1`] = `
                 class="default_xu2jcg-o_O-menuWrapper_wvrnr4"
               >
                 <button
-                  aria-expanded="true"
+                  aria-expanded="false"
```
Comment (kevinbarabash) [inline packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap:389]: Why did this change?

File: packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap [hunk 2]
```diff
@@ -395,7 +395,7 @@ exports[`renderer snapshots correct answer: correct answer 1`] = `
                   <span
                     class="text_f1191h-o_O-LabelMedium_1rew30o-o_O-text_16ujc3c"
                   >
-                    greater/less than or equal to
+                    less than or equal to
```
Comment (kevinbarabash) [inline packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap:398]: Why did this change?

File: packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap [hunk 3]
```diff
@@ -457,7 +457,7 @@ exports[`renderer snapshots incorrect answer: incorrect answer 1`] = `
                 class="default_xu2jcg-o_O-menuWrapper_wvrnr4"
               >
                 <button
-                  aria-expanded="true"
+                  aria-expanded="false"
```
Comment (kevinbarabash) [inline packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap:460]: Why did this change?

File: packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap [hunk 4]
```diff
@@ -466,7 +466,7 @@ exports[`renderer snapshots incorrect answer: incorrect answer 1`] = `
                   <span
                     class="text_f1191h-o_O-LabelMedium_1rew30o-o_O-text_16ujc3c"
                   >
-                    greater/less than or equal to
+                    greater than or equal to
```
Comment (kevinbarabash) [inline packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap:469]: Why did this change?

File: packages/perseus/src/components/__tests__/zoomable_test.jsx [hunk 5]
```diff
@@ -133,7 +133,7 @@ describe("Zoomable", () => {
         expect(container).toMatchInlineSnapshot(`
             <div>
               <span
-                style="display: block; width: 100%; transform: scale(0.4993757802746567, 0.4993757802746567); transform-origin: 0 0; opacity: 1; height: 101px; transition-property: opacity transform; transition-duration: 0.3s; transition-timing-function: ease-out;"
+                style="display: block; width: 100%; transform: scale(0.4993757802746567, 0.4993757802746567)  translate(0, 8px); transform-origin: 0 0; opacity: 0; height: 101px;"
```
Comment (kevinbarabash) [inline packages/perseus/src/components/__tests__/zoomable_test.jsx:136]: Why did this change?

**Review by kevinbarabash — COMMENTED** [review]
I'm glad that the tests are running faster now, but I'm concerned that the snapshots have changed.

File: packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap (hunk 1 above)
Comment (jeremywiebe) [reply packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap:389]: This is the correct value. The snapshot in the original PR from webapp had this wrong and I didn't notice it (the dropdown is in a closed state)

File: packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap (hunk 2 above)
Comment (jeremywiebe) [reply packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap:398]: This is the correct value. The snapshot in the original PR from webapp had this wrong and I didn't notice it (the dropdown is in a closed state)

File: packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap (hunk 3 above)
Comment (jeremywiebe) [reply packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap:460]: This is the correct value. The snapshot in the original PR from webapp had this wrong and I didn't notice it (the dropdown is in a closed state)

File: packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap (hunk 4 above)
Comment (jeremywiebe) [reply packages/perseus/src/__tests__/__snapshots__/renderer_test.jsx.snap:469]: This is the correct value. The snapshot in the original PR from webapp had this wrong and I didn't notice it (this is the selected value after closing the dropdown)

**Review by kevinbarabash — APPROVED** [review]
Thanks for replying to all my comments about snapshots changing.  I'm glad that we have the right values now. :shipit:

File: packages/perseus/src/components/__tests__/zoomable_test.jsx (hunk 5 above)
Comment (jeremywiebe) [reply packages/perseus/src/components/__tests__/zoomable_test.jsx]: Turns out the test needed a second call to `jest.runOnlyPendingTimers()`. I'm not sure why it worked in the webapp repo... I suspect a subtle difference in the jest setup?

File: config/test/test-setup.js [hunk 6]
```diff
@@ -3,8 +3,6 @@
  * This file is loaded after the jest test framework has be initialized
  * but before any tests have run.
  */
-import React16EnzymeAdapter from "enzyme-adapter-react-16";
-import Enzyme from "enzyme"; // eslint-disable-line no-restricted-imports
```
Comment (jeremywiebe) [inline config/test/test-setup.js:7]: These lines got duplicated during one of my earlier PR merges.

**Pull request merged**